crochet/_shutdown.py

```diff
@@ -53,8 +54,15 @@ def run(self):
                 log.err()
 
 
-# This is... fragile. Not sure how else to do it though.
-_registry = FunctionRegistry()
-_watchdog = Watchdog([t for t in threading.enumerate()
-                     if t.name == "MainThread"][0], _registry.run)
-register = _registry.register
+def default_shutdown_condition():
```

Comment:
I may be forgetting why this was necessary though, so if you think it's too big of a change for you (or this branch) it's not necessary to do this.